### PR TITLE
pivx-cli better error handling + libevent RAII upstream backports

### PR DESCRIPTION
**Pull request by furszy**
Backported the following PRs from upstream:

* #8722.
* #9265.
* #9073.
* #9387.
* #9517.
* #9647.
* #13394.

### CODE REVIEW

**Review by random-zebra — DISMISSED** [review]
utACK 96991e630a1bd263109aa2e56a46dd4936e4b713

File: src/test/raii_event_tests.cpp [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    free(mem);
+}
+
+BOOST_FIXTURE_TEST_SUITE(raii_event_tests, BasicTestingSetup)
+
+BOOST_AUTO_TEST_CASE(raii_event_creation)
+{
+    event_set_mem_functions(tag_malloc, realloc, tag_free);
+
+    void* base_ptr = NULL;
```
[the remark this answers is not in the document]
Comment (furszy) [reply src/test/raii_event_tests.cpp]: pushed.

**Comment by furszy** [comment]
squashed feedback in 0d4e7ae.

**Review by random-zebra — APPROVED** [review]
re-utACK b4de8aa9998a7542cd33463122c5734be64c9662

**Review by Fuzzbawls — APPROVED** [review]
ACK b4de8aa9998a7542cd33463122c5734be64c9662

**Pull request merged**